subsys/nonsecure/Kconfig

```diff
@@ -0,0 +1,273 @@
+
+menu "Peripheral Secure mapping"
+
+config NRF_FPU_SECURE
+	bool "FPU configured as secure peripheral"
+	depends on CPU_HAS_FPU
+	help
+	  FPU is implemented with user-selectable security mapping.
+          If it has been configured as a secure peripheral then only the secure
+          processing environment is able to access the peripheral
+
+peripheral=DCNF
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=OSCILLATORS
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=REGULATORS
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=CLOCK
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=POWER
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=RESET
+rsource "Kconfig.template.peripheral_secure"
+
+config NRF_CTRLAP_SECURE
+	bool "CTRLAP configured as secure peripheral"
+	# TODO: Use template when HAS_HW_NRF_CTRLAP is available in zephyr.
+	depends on SOC_NRF9160 || SOC_NRF5340_CPUAPP
+	help
+	  CTRLAP is implemented with user-selectable security mapping.
+          If it has been configured as a secure peripheral then only the secure
+          processing environment is able to access the peripheral
+
+peripheral=SPIM0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIS0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIM0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIS0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=UARTE0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIM1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIS1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIM1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIS1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=UARTE1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIM4
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIM2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIS2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIM2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIS2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=UARTE2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIM3
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SPIS3
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIM3
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TWIS3
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=UARTE3
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=SAADC
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TIMER0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TIMER1
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=TIMER2
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=RTC0
+rsource "Kconfig.template.peripheral_secure"
+
+peripheral=RTC1
+rsource "Kconfig.template.peripheral_secure"
+
+config NRF_DPPI_SECURE
```

Comment:
Why not use the template also here? The peripheral is in fact named DPPIC, so `NRF_DPPIC_SECURE` would be an appropriate option for it.

Reply:
Well, since both are being used I prefer to use the one that introduces the least amount of naming differences to the user. And that means aligning them to the nrfx naming. For example:
```
config NRFX_DPPI
	bool "Enable DPPI allocator"
	depends on HAS_HW_NRF_DPPIC
```

Reply:
When it comes to the peripheral itself, only DPPIC is used. That's why the `HAS_HW_NRF_DPPIC` option has this C at the end and why there is a `dppic` node in devicetree, not `dppi`. The above option enables compilation of the allocator for DPPI channels and groups, hence it uses DPPI, not DPPIC.
But it's actually a detail, so I won't insist on this. If you prefer to keep it as it is now, I'm fine with that.

Reply:
I prefer to keep it, it's not only my preference though, since I've already decided to use NRF_DPPI_ in TF-M changing it now would cause an inconsistency there as well.